streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -296,6 +296,75 @@ public void shouldAddTasksToStateUpdater() {
         Mockito.verify(stateUpdater).add(task01);
     }
 
+    @Test
+    public void shouldHandleRemovedTasksFromStateUpdater() {
```

Comment:
The exceptional cases are missing like `newTask.initializeIfNeeded();` throws an exception or closing a tasks throws an exception.

Reply:
When adding tests I realized we do not need to handle exceptions by closing tasks dirty and then re-thrown, but could simply re-throw directly and delegate to the StreamThread to handle the thrown exception. Will updated accordingly.